### ISSUE TITLE
Update the softmax issue with DNA encoder.

### ISSUE THREAD

**Issue by zmgong** [bioscan-ml/clibd#33]
The forward function of the DNA encoder is mistakenly using:
`return self.base_dna_encoder(sequence).logits.softmax(dim=-1).mean(dim=1)`
The correct way should be using:
`return self.lora_barcode_bert(x).hidden_states[-1].mean(dim=1)`

To fix this, the idea is to create a new branch, re-train all existed pre-trained models to update relevent checkpoints. Then merge to the main branch.
